File: core/src/main/java/io/github/therealsegfault/projectbeatsgdx/core/Lane.java
package io.github.therealsegfault.projectbeatsgdx.core;

public enum Lane {
    W(0), A(1), S(2), D(3);

    public final int id;
    Lane(int id) { this.id = id; }

    public static Lane fromId(int id) {
        return switch (id) {
            case 0 -> W;
            case 1 -> A;
            case 2 -> S;
            case 3 -> D;
            default -> throw new IllegalArgumentException("Invalid lane id: " + id);
        };
    }
}

File: core/src/main/kotlin/io/github/therealsegfault/projectbeatsgdx/core/Lane.kt
package io.github.therealsegfault.projectbeatsgdx.core

enum class Lane(val id: Int) {
  UP(0),
  LEFT(1),
  DOWN(2),
  RIGHT(3);

  companion object {
    fun fromId(id: Int): Lane {
      return when (id) {
        0 -> UP
        1 -> LEFT
        2 -> DOWN
        3 -> RIGHT
        else -> RIGHT
      }
    }
  }
}
